src/pos/erc20.ts

```diff
@@ -151,6 +151,34 @@ export class ERC20 extends POSToken {
         });
     }
 
+    private depositEtherWithGas_(amount: TYPE_AMOUNT, userAddress: string, swapEthAmount: TYPE_AMOUNT, swapCallData: string, option: ITransactionOption = {}) {
+        this.checkForRoot("depositEtherWithGas");
+
+        return this.getChainId().then((chainId: number) => {
+            if (chainId !== 1) {
```

Comment:
what if I am testing on goerli(5), should it tolerate testnet also?

Reply:
its not possible on testnet.

Reply:
feature is only for mainnet